### PR TITLE
MENFORCER-383 add rulesFile user property to supply enforcer rules from cmd line

### PR DESCRIPTION
**Pull request by marioja**
MENFORCER-383  The rulesFile is better the the rules user property because it is
possible to supply complete rules with arguments etc.

Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MENFORCER) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [x ] Each commit in the pull request should have a meaningful subject line and body.
 - [x ] Format the pull request title like `[MENFORCER-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MENFORCER-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [x ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [ x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by marioja** [comment]
I added second commit to handle recursion loop better.  On second executeMojo call I remove the rulesFilename from the configuration.  This has the benefit of allowing the use of rulesFilename along with rules configured in the  inheritance and merging them together.

**Comment by marioja** [comment]
Added an IT test for the rulesFile user property.

**Comment by marioja** [comment]
This pr is ready for approval.  Can someone look into it.

**Comment by marioja** [comment]
ping?

File: maven-enforcer-plugin/src/main/java/org/apache/maven/plugins/enforcer/EnforceMojo.java [hunk 1]
```diff
@@ -150,6 +175,53 @@ public void execute()
...
+                {
+                    configuration.removeChild( i ); // remove to avoid re-doing this if block
+                    break; // done exit
+                }
+            }
+            MojoExecution mojoExecution = new MojoExecution( mojoDescriptor, configuration );
+            
+            try
+            {
+                pluginManager.executeMojo( session, mojoExecution );
```
Comment (rfscholte) [inline maven-enforcer-plugin/src/main/java/org/apache/maven/plugins/enforcer/EnforceMojo.java:215]: This is some kind of inception that doesn't make sense here. There is a better way to solve these kind of problems: maven-extensions.
I've implemented MENFORCER-384, which makes commandlineRules absolete.

**Comment by rfscholte** [comment]
I'm sorry that you've spend some time on this, but this was not the right approach (although I do understand your thinking).  This PR has been superseded with the implementation MENFORCER-384.

**Comment by jira-importer** [comment]
Resolve #496

**Pull request closed without merging**